### PR TITLE
LPS-183125 Adding Users Page Restriction

### PR DESCRIPTION
**Pull request by EugenioAntunes**

### CODE REVIEW

**Review by rafaelfs98 — APPROVED** [review]
Nice Joob man :fire:

**Review by Elvinho — APPROVED** [review]
Nice 💾

**Review by welitonpe — APPROVED** [review]
Good Job!!! :fire: :fire: :fire:

**Review by alvarolealdl — APPROVED** [review]
Good Job @EugenioAntunes

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx [hunk 1]
```diff
@@ -110,7 +110,7 @@ const CaseListView: React.FC<CaseListViewProps> = ({
 				...tableProps,
 			}}
 			transformData={(response) =>
-				testrayCaseRest.transformDataFromList(response)
+				testrayCaseRestImpl.transformDataFromList(response)
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx:113]:
```suggestion
				testrayCaseImpl.transformDataFromList(response)
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayRequirement.ts [hunk 2]
```diff
@@ -12,73 +12,55 @@
...
+
 import yupSchema from '../../schema/yup';
-import fetcher from '../fetcher';
-import {APIResponse, TestrayRequirement} from './types';
+import {TestrayRequirement} from './types';
 
 type Requirement = typeof yupSchema.requirement.__outputType & {
 	projectId: number;
 };
+class TestrayRequirementsImpl extends Rest<Requirement, TestrayRequirement> {
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayRequirement.ts:23]: Add space

```suggestion

class TestrayRequirementsImpl extends Rest<Requirement, TestrayRequirement> {
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestraySuite.ts [hunk 3]
```diff
@@ -76,4 +76,6 @@ class TestraySuiteRest extends Rest<Suite, TestraySuite> {
 	}
 }
 
-export const testraySuiteRest = new TestraySuiteRest();
+const testraySuiteImpl = new TestraySuiteRest();
+
+export {testraySuiteImpl};
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestraySuite.ts:81]:
```suggestion
const testraySuiteImpl = new TestraySuiteImpl();

export {testraySuiteImpl};
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
@EugenioAntunes thank you for the hard work, it looks really good!

the suggestions I made was just `typo` to keep the consistence to the project

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/CompareRuns/CompareRunsCases.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ import {StatusBadgeType} from '~/components/StatusBadge/StatusBadge';
 import useSearchBuilder from '~/hooks/useSearchBuilder';
 import i18n from '~/i18n';
 import {TestrayCase, TestrayCaseResult, TestrayRun} from '~/services/rest';
-import {testrayCaseRest} from '~/services/rest/TestrayCase';
+import {testrayCaseRestImpl} from '~/services/rest/TestrayCase';
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/CompareRuns/CompareRunsCases.tsx:24]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/CompareRuns/CompareRunsCases.tsx [hunk 5]
```diff
@@ -168,7 +168,7 @@ const CompareRunsCases = () => {
 						filterSchema: 'compareRunsCases',
 					}}
 					resource={`/${
-						testrayCaseRest.uri
+						testrayCaseRestImpl.uri
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/CompareRuns/CompareRunsCases.tsx:171]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseForm.tsx [hunk 6]
```diff
@@ -31,7 +32,7 @@ import {
 	TestrayCaseType,
 	TestrayComponent,
 	TestrayProject,
-	testrayCaseRest,
+	testrayCaseRestImpl,
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseForm.tsx:35]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Suites/Suite.tsx [hunk 7]
```diff
@@ -25,7 +25,7 @@ import {
 	TestraySuite,
 	TestraySuiteCase,
 	suitesCasesResource,
-	testrayCaseRest,
+	testrayCaseRestImpl,
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Suites/Suite.tsx:28]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayCase.ts [hunk 8]
```diff
@@ -102,4 +102,6 @@ class TestrayCaseRest extends Rest<Case, TestrayCase> {
 	}
 }
 
-export const testrayCaseRest = new TestrayCaseRest();
+const testrayCaseRestImpl = new TestrayCaseRest();
+
+export {testrayCaseRestImpl};
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/services/rest/TestrayCase.ts:107]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseForm.tsx [hunk 9]
```diff
@@ -126,8 +127,8 @@ const CaseForm = () => {
 		onSubmit(
 			{...form, projectId},
 			{
-				create: (data) => testrayCaseRest.create(data),
-				update: (id, data) => testrayCaseRest.update(id, data),
+				create: (data) => testrayCaseRestImpl.create(data),
+				update: (id, data) => testrayCaseRestImpl.update(id, data),
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseForm.tsx:131]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseForm.tsx [hunk 10]
```diff
@@ -279,4 +280,6 @@ const CaseForm = () => {
 	);
 };
 
-export default CaseForm;
+export default withPagePermission(CaseForm, {
+	restImpl: testrayCaseRestImpl,
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseForm.tsx:284]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseOutlet.tsx [hunk 11]
```diff
@@ -43,10 +43,10 @@ const CaseOutlet = () => {
 	}: {testrayProject: TestrayProject} = useOutletContext();
 
 	const {data: testrayCase, mutate} = useFetch<TestrayCase>(
-		testrayCaseRest.getResource(caseId as string),
+		testrayCaseRestImpl.getResource(caseId as string),
 		{
 			transformData: (response) =>
-				testrayCaseRest.transformData(response),
+				testrayCaseRestImpl.transformData(response),
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/CaseOutlet.tsx:49]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/Export.tsx [hunk 12]
```diff
@@ -259,7 +259,7 @@ const Export = () => {
 			},
 			swrConfig: {shouldFetch: caseIds.length},
 			transformData: (response) =>
-				testrayCaseRest.transformDataFromList(response),
+				testrayCaseRestImpl.transformDataFromList(response),
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/Export.tsx:262]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx [hunk 13]
```diff
@@ -21,7 +21,7 @@ import {ListViewContextProviderProps} from '../../../context/ListViewContext';
 import SearchBuilder from '../../../core/SearchBuilder';
 import {FormModal} from '../../../hooks/useFormModal';
 import i18n from '../../../i18n';
-import {testrayCaseRest} from '../../../services/rest';
+import {testrayCaseRestImpl} from '../../../services/rest';
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx:24]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx [hunk 14]
```diff
@@ -56,7 +56,7 @@ const CaseListView: React.FC<CaseListViewProps> = ({
 				filterSchema: 'cases',
 				title: i18n.translate('cases'),
 			}}
-			resource={testrayCaseRest.resource}
+			resource={testrayCaseRestImpl.resource}
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx:59]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx (hunk 1 above)
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Cases/index.tsx:113]: Done!

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx [hunk 15]
```diff
@@ -330,4 +331,6 @@ const UserForm = () => {
 	);
 };
 
-export default UserForm;
+export default withPagePermission(UserForm, {
+	restImpl: liferayUserAccountsImpl,
+});
```
[the remark this answers is not in the document]
Comment (EugenioAntunes) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Manage/User/UserForm.tsx:336]: Done!

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Well done @EugenioAntunes

**Comment by kevenleone** [comment]
resent: https://github.com/pt-liferay-solutions/liferay-portal/pull/3870#issuecomment-1528041725

**Pull request closed without merging**